staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -781,11 +794,21 @@ func (qs *queueSet) finishRequestAndDispatchAsMuchAsPossible(req *request) bool
 func (qs *queueSet) finishRequestLocked(r *request) {
 	now := qs.clock.Now()
 	qs.totRequestsExecuting--
-	qs.totSeatsInUse -= r.Seats()
 	metrics.AddRequestsExecuting(r.ctx, qs.qCfg.Name, r.fsName, -1)
-	metrics.AddRequestConcurrencyInUse(qs.qCfg.Name, r.fsName, -r.Seats())
 	qs.obsPair.RequestsExecuting.Add(-1)
 
+	defer func() {
+		// if the request has additional latency then we need to wait
+		// before we free up the these seats
+		if r.width.AdditionalLatency <= 0 {
+			qs.totSeatsInUse -= r.Seats()
```

Comment:
This isn't sleeping yet...

Reply:
OK - I see now that this is desired.
I think this requires a more extensive comments to describe what exactly is happening here.